hazelcast/src/main/java/com/hazelcast/internal/metrics/impl/FieldProbe.java

```diff
@@ -28,30 +31,18 @@
 import java.util.Map;
 import java.util.concurrent.Semaphore;
 
-import static com.hazelcast.internal.metrics.impl.ProbeUtils.TYPE_COLLECTION;
-import static com.hazelcast.internal.metrics.impl.ProbeUtils.TYPE_COUNTER;
-import static com.hazelcast.internal.metrics.impl.ProbeUtils.TYPE_DOUBLE_NUMBER;
-import static com.hazelcast.internal.metrics.impl.ProbeUtils.TYPE_DOUBLE_PRIMITIVE;
-import static com.hazelcast.internal.metrics.impl.ProbeUtils.TYPE_LONG_NUMBER;
-import static com.hazelcast.internal.metrics.impl.ProbeUtils.TYPE_MAP;
-import static com.hazelcast.internal.metrics.impl.ProbeUtils.TYPE_PRIMITIVE_LONG;
-import static com.hazelcast.internal.metrics.impl.ProbeUtils.TYPE_SEMAPHORE;
-import static com.hazelcast.internal.metrics.impl.ProbeUtils.getType;
-import static com.hazelcast.internal.metrics.impl.ProbeUtils.isDouble;
-import static java.lang.String.format;
-
 /**
  * A FieldProbe is a {@link ProbeFunction} that reads out a field that is annotated with {@link Probe}.
  */
 abstract class FieldProbe implements ProbeFunction {
 
     final CachedProbe probe;
     final Field field;
-    final int type;
+    final ProbeUtils type;
     final SourceMetadata sourceMetadata;
     final String probeName;
 
-    FieldProbe(Field field, Probe probe, int type, SourceMetadata sourceMetadata) {
+    FieldProbe(Field field, Probe probe, ProbeUtils type, SourceMetadata sourceMetadata) {
```

Comment:
ProbeUtils can't be a type :) So some cleanup is needed.

Reply:
> ProbeUtils can't be a type :) So some cleanup is needed.

Why can't `ProbeUtils` be a type? I've converted it from a bunch of `int` constants to an enum as per the PR description.

Reply:
https://github.com/hazelcast/hazelcast/pull/25279#pullrequestreview-1642027251